plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go

```diff
@@ -160,25 +164,51 @@ func (d *defaultServicePlan) Validate() error {
 	if d.scClient == nil {
 		return errors.New("missing service class interface")
 	}
+	if d.spClient == nil {
+		return errors.New("missing serviceplan interface")
+	}
 	if d.spLister == nil {
 		return errors.New("missing service plan lister")
 	}
 	return nil
 }
 
-func (d *defaultServicePlan) getServiceClassByExternalName(a admission.Attributes, scName string) (*servicecatalog.ClusterServiceClass, error) {
-	glog.V(4).Infof("Fetching serviceclass as %q", scName)
-	listOpts := apimachineryv1.ListOptions{FieldSelector: "spec.externalName==" + scName}
-	servicePlans, err := d.scClient.List(listOpts)
+func (d *defaultServicePlan) getClusterServiceClassByExternalName(a admission.Attributes, scName string) (*servicecatalog.ClusterServiceClass, error) {
+	glog.V(4).Infof("Fetching serviceclass filtered by class external name %q", scName)
+	fieldSet := fields.Set{
```

Comment:
Just curious why replace a single line with 4?

Reply:
This method escapes the field selector if necessary

Reply:
I picked up this initial change from Paul and I had to ask him.  He says this "escapes the field selector if necessary" and its a best practice per api machinery guys